frontend/packages/ux-editor/src/components/config/editModal/EditTextResourceBindings.tsx

```diff
@@ -16,33 +17,56 @@ export const EditTextResourceBindings = ({
   handleComponentChange,
   textResourceBindingKeys,
 }: EditTextResourceBindingsProps) => {
-  const t = useText();
-  const [open, setOpen] = React.useState(true);
+  const { t } = useTranslation();
 
-  const toggleAccordion = () => {
-    setOpen(!open);
+  const [keysSet, setKeysSet] = React.useState(
+    Object.keys(component.textResourceBindings || {}),
+  );
+
+  const [keysToAdd, setKeysToAdd] = React.useState<string[]>(textResourceBindingKeys.filter((key) => !keysSet.includes(key)));
+
+  useEffect(() => {
+    setKeysToAdd(textResourceBindingKeys.filter((key) => !keysSet.includes(key)));
+  }, [keysSet, setKeysToAdd, textResourceBindingKeys]);
```

Comment:
I think we can remove the useEffect, useState and avoid the duplicated `textResourceBindingKeys.filter`, by refactoring the code to look something like this?
```suggestion
  const keysToAdd = useMemo(() => textResourceBindingKeys.filter((key) => !keysSet.includes(key)), [keysSet, textResourceBindingKeys])
```